### PR TITLE
2655/search listings

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue Overview

This PR addresses #2655 #2656

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR refactors:
- listing table
- applications table
- users table

I created a reusable solution and implemented it in all places (except flagged applications, it's a bit different, and should be refactored in the future). The new table component also includes search input.

## How Can This Be Tested/Reviewed?

1. Open partners and login
2. Validate listings table (test search input, pagination, sorting)
3. Do the same for applications and users table

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by seanmalbert — APPROVED** [review]
Looks good @dominikx96, once the cypress-partners test is passing. I created @2720, for @pbn4 to add a fuzzy search for name.

**Review by seanmalbert — CHANGES_REQUESTED** [review]
Address failing tests.

**Comment by dominikx96** [comment]
@seanmalbert I noticed there is a problem with listing tests. That's because we're testing a page immediately, but a listing table needs some time to finish loading. I added a 5000ms delay to make sure that results are in place, hope it will solve this issue.

**Comment by dominikx96** [comment]
@seanmalbert I added 2 changes to check if it fixes tests, but please don't merge until tomorrow morning. I want to revert the default cypress timeout and determine which part of the change fails tests.

**Comment by dominikx96** [comment]
@seanmalbert I decided to refactor all tables (except flagged apps, there is a missing search param backend side) because the current code was repeated in a couple of places.

File: sites/partners/cypress/integration/05-paperApplication.spec.ts
Comment (emilyjablonski) [inline sites/partners/cypress/integration/05-paperApplication.spec.ts:12]: If possible, we want to get items by test ids to rely less on implementation details. If we changed the row-id here in the component, it would likely break the test, and potentially cause a false negative. It's clearer in the components to use a test id so we know where to put it if we do a refactor.

File: ui-components/src/tables/AgTable.tsx
Comment (emilyjablonski) [inline ui-components/src/tables/AgTable.tsx:121]: 🧹

**Comment by emilyjablonski** [comment]
Screen Shot 2022-05-25 at 10 27 44 AM
Screen Shot 2022-05-25 at 10 25 26 AM
Screen Shot 2022-05-25 at 10 24 55 AM

File: sites/partners/cypress/integration/05-paperApplication.spec.ts
Comment (dominikx96) [reply sites/partners/cypress/integration/05-paperApplication.spec.ts:12]: Make sense! I changed this while testing broken cypress tests, agree with you, restoring the previous declaration, thanks!

**Comment by dominikx96** [comment]
Thanks, @emilyjablonski for the detailed review! I also think we need more reviews before merging.
I've updated:
- fixed problems listed above
- mobile version, now it should work better!
- added `AgGrid` story
- added ui tests!

**Review by seanmalbert — APPROVED** [review]
looks good!

**Pull request merged**
